### DISCUSSION TITLE
Should the XBEGIN instruction have a branch type?

### DISCUSSION THREAD

**Discussion by jonomango** [zyantific/zydis#352] [Q&A] [answered]
Hello,

I've noticed that the `XBEGIN` instruction has `meta.branch_type` set to `ZYDIS_BRANCH_TYPE_NONE` when decoded, even though it is considered a conditional branch (`meta.category` is set to `ZYDIS_CATEGORY_COND_BR`). I understand that there currently isn't a branch type that would fit for `XBEGIN`, so I was wondering if this was an intentional design choice or maybe specified in the Intel SDM somewhere.

**Comment by flobernd** [comment, marked as answer]
Hey @jonomango,

I think you found a bug and I actually just forgot to mark this instruction as a "near branch" 🙂

Will create an issue for that and fix in the next days. Thanks for reporting!
